src/validator.ts

```diff
@@ -27,8 +27,7 @@ export const validate = async (publicKey: PublicKey, entry: IPNSEntry) => {
 
     validateCborDataMatchesPbData(entry)
   } else {
-    signature = entry.signature ?? new Uint8Array(0)
-    dataForSignature = ipnsEntryDataForV1Sig(value, validityType, validity)
+    throw errCode(new Error('missing data or signatureV2'), ERRORS.ERR_SIGNATURE_VERIFICATION)
```

Comment:
Maybe split `if ((entry.signatureV2 != null) && (entry.data != null)) {` with an escape hatch thrown error for missing data, and one for signature, instead of combining them?

Reply:
the more we combine errors, and the less detail we provide, the more problems we will run into like ipfs/interop#462 where it's much harder to track down issues than it needs to be.

Reply:
I prefer to keep it simple and have single error: these fields are all-or-nothing.